tools/matrix-echo

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env bash
+
+scan_args() {
+   for each in "$@"; do
+      if [[ "$each" == "-u" || "$each" == "--user" ]]; then
+         USER="$each"
+      fi
+      if [[ "$each" == "-p" || "$each" == "--password" ]]; then
+         PASSWORD="$each"
+      fi
+   done
+}
+
+cleanup() {
+   sleep 1
+   rm -f "$CONFIG" 2>/dev/null
+}
+
+trap cleanup SIGINT SIGHUP EXIT
+
+scan_args
+
+# Try to fetch credentials from .git-credentials.
+if [[ -f "$HOME/.git-credentials" ]]; then
+   line=$(grep "gitlab.igalia.com" ~/.git-credentials)
+   if [[ "$?" -eq "0" ]]; then
+      credentials=$(echo "$line" | sed -r 's|https?://(\w+):(\w+)@gitlab.igalia.com|\1:\2|g')
```

Comment:
Could be this domain an parameter like the credentials are?

Reply:
Yep, I can search for the 'domain' in the template config.